torch/csrc/jit/python/init.cpp

```diff
@@ -123,6 +126,81 @@ using ::c10::FunctionSchema;
 using caffe2::serialize::PyTorchStreamReader;
 using caffe2::serialize::PyTorchStreamWriter;
 
+class PythonSymbolicIntNode: public c10::SymbolicIntNode {
+public:
+  PythonSymbolicIntNode(py::object pyobj):
+    c10::SymbolicIntNode() {
+      pyobj_ = std::make_shared<c10::SafePyObject>(pyobj.release().ptr(), getPyInterpreter());
+    };
+
+  virtual SymbolicIntNode* wrap(int64_t num) {
+    auto r = getPyObj().attr("wrap")(num);
+    return new PythonSymbolicIntNode(r);
+  }
+
+  virtual bool bool_() {
+    py::gil_scoped_acquire acquire;
+    return py::str(getPyObj().attr("__bool__")()).is(py::str(Py_True));
```

Comment:
why are you doing a string comparison to test what the bool result is?

Reply:
lemme try simplifying it a bit. This what SO recommends to do, but I do agree it's convoluted. There's no py::cast to `bool` but maybe we don't have to do `py::str` on both sides.

Reply:
cleaned up this part. Now it should make more sense.